Shared library build failure on macOS

Building the shared library with
```sh
make BUILD_MODULE=OFF BUILD_SHARED_LIBS=ON
```
fails on macOS. I am running macOS 11.1 and using the system compiler (Clang 12). Building the module and the static library succeeds.

The build error reads
```
Undefined symbols for architecture x86_64:
  "_luaL_argerror", referenced from:
      _luv_loop_configure in luv.c.o
      _luv_cancel in luv.c.o
      _luv_req_get_type in luv.c.o
      _luv_timer_start in luv.c.o
      _luv_timer_stop in luv.c.o
      _luv_timer_again in luv.c.o
      _luv_timer_set_repeat in luv.c.o
      ...
[snip]
  "_lua_typename", referenced from:
      _luv_stack_dump in luv.c.o
      _luv_check_callable in luv.c.o
      _luv_thread_arg_push in luv.c.o
      _luv_thread_arg_set in luv.c.o
      _luv_check_bufs in luv.c.o
      _luv_prep_bufs in luv.c.o
      _luv_check_bufs_noref in luv.c.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [libluv.1.36.0.dylib] Error 1
make[2]: *** [CMakeFiles/libluv.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [luv] Error 2
```
Full build output: [build.log](https://github.com/luvit/luv/files/5836256/build.log)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,6 +197,10 @@ if (BUILD_SHARED_LIBS)
   set_target_properties(libluv PROPERTIES PREFIX "")
   set_target_properties(libluv
     PROPERTIES VERSION ${LUV_VERSION} SOVERSION ${LUV_VERSION_MAJOR})
+  if(APPLE)
+    set_target_properties(libluv
+      PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+  endif(APPLE)
   list(APPEND ACTIVE_TARGETS "libluv")
 endif (BUILD_SHARED_LIBS)
 

